Fails to build docs and man with GDAL 3.1

## Expected behavior and actual behavior.

`make docs` fails, whereas it should succeed.

## Steps to reproduce the problem.

`make docs` returns:
```
[  818s] + make -j1 VERBOSE=1 docs
[  818s] (cd doc; make html)
[  818s] make[1]: Entering directory '/home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc'
[  818s] python3 "source/build_driver_summary.py" "source/drivers/raster" raster_driver_summary "source/drivers/raster/driver_summary.rst"
[  819s] python3 "source/build_driver_summary.py" "source/drivers/vector" vector_driver_summary "source/drivers/vector/driver_summary.rst"
[  819s] python3 "source/build_configoptions_index.py"  "source" "source/user/configoptions_index_generated.rst"
[  819s] sphinx-build -M html "source" "build" --keep-going -j auto -W 
[  820s] Running Sphinx v2.3.1
[  821s] making output directory... done
[  821s] building [mo]: targets for 0 po files that are out of date
[  821s] building [html]: targets for 459 source files that are out of date
[  821s] updating environment: [new config] 459 added, 0 changed, 0 removed
[  821s] reading sources... [  3%] about .. api/gnm_cpp
[  821s] /usr/lib/python3.8/site-packages/sphinx/util/docutils.py:283: RemovedInSphinx30Warning: function based directive support is now deprecated. Use class based directive instead.
[  821s]   warnings.warn('function based directive support is now deprecated. '
[  821s] reading sources... [  6%] api/index .. development/index
[  821s] reading sources... [ 10%] development/rfc/index .. development/rfc/rfc23_ogr_unicode
[  821s] reading sources... [ 13%] development/rfc/rfc24_progressive_data_support .. development/rfc/rfc38_ogr_faster_open
[  821s] reading sources... [ 17%] development/rfc/rfc39_ogr_layer_algebra .. development/rfc/rfc52_strict_sql_quoting
[  821s] reading sources... [ 20%] development/rfc/rfc53_ogr_notnull_default .. development/rfc/rfc66_randomlayerreadwrite
[  821s] reading sources... [ 24%] development/rfc/rfc67_nullfieldvalues .. drivers/raster/Idrisi
[  822s] reading sources... [ 27%] drivers/raster/aaigrid .. drivers/raster/cals
[  822s] reading sources... [ 31%] drivers/raster/ceos .. drivers/raster/doq2
[  822s] reading sources... [ 34%] drivers/raster/driver_summary .. drivers/raster/fit
[  822s] reading sources... [ 37%] drivers/raster/fits .. drivers/raster/gta
[  822s] 
[  822s] Sphinx parallel build error:
[  822s] breathe.directives.MTimerError: Cannot find file: /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/build/xml/index.xml
```

The same happens with `man`:
```
[  804s] + make -j1 VERBOSE=1 man
[  804s] (cd doc; make man)
[  804s] make[1]: Entering directory '/home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc'
[  804s] python3 "source/build_driver_summary.py" "source/drivers/raster" raster_driver_summary "source/drivers/raster/driver_summary.rst"
[  804s] python3 "source/build_driver_summary.py" "source/drivers/vector" vector_driver_summary "source/drivers/vector/driver_summary.rst"
[  804s] python3 "source/build_configoptions_index.py"  "source" "source/user/configoptions_index_generated.rst"
[  804s] sphinx-build -M man "source" "build" --keep-going -j auto -W 
[  805s] Running Sphinx v2.3.1
[  806s] making output directory... done
[  806s] building [mo]: targets for 0 po files that are out of date
[  806s] building [man]: all manpages
[  806s] updating environment: [new config] 459 added, 0 changed, 0 removed
[  806s] reading sources... [  3%] about .. api/gnm_cpp
[  806s] /usr/lib/python3.8/site-packages/sphinx/util/docutils.py:283: RemovedInSphinx30Warning: function based directive support is now deprecated. Use class based directive instead.
[  806s]   warnings.warn('function based directive support is now deprecated. '
[  806s] reading sources... [  6%] api/index .. development/index
[  806s] reading sources... [ 10%] development/rfc/index .. development/rfc/rfc23_ogr_unicode
[  806s] reading sources... [ 13%] development/rfc/rfc24_progressive_data_support .. development/rfc/rfc38_ogr_faster_open
[  806s] reading sources... [ 17%] development/rfc/rfc39_ogr_layer_algebra .. development/rfc/rfc52_strict_sql_quoting
[  806s] reading sources... [ 20%] development/rfc/rfc53_ogr_notnull_default .. development/rfc/rfc66_randomlayerreadwrite
[  806s] reading sources... [ 24%] development/rfc/rfc67_nullfieldvalues .. drivers/raster/Idrisi
[  806s] reading sources... [ 27%] drivers/raster/aaigrid .. drivers/raster/cals
[  806s] reading sources... [ 31%] drivers/raster/ceos .. drivers/raster/doq2
[  806s] 
[  806s] Sphinx parallel build error:
[  806s] breathe.directives.MTimerError: Cannot find file: /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/build/xml/index.xml
```

## Operating system

openSUSE Tumbleweed and Leap 15.1

## GDAL version and provenance

GDAL 3.1.0

`doc/build/xml/` is empty.
If I re-generate the Doxygen doc as requested, the `xml/` folder is populated but I get the following error:
```
[  123s] Package newunicodechar Warning: Redefining Unicode character on input line 7.
[  123s] 
[  123s] 
[  123s] Package newunicodechar Warning: Redefining Unicode character on input line 8.
[  123s] 
[  123s] (/usr/share/texmf/tex/latex/l3backend/l3backend-dvips.def)
[  123s] No file _formulas.aux.
[  123s] [1] [2] [3] [4] [5] [6] [7] [8] [9] [10] [11] [12] [13] [14] [15] [16] [17]
[  123s] [18] [19] (./_formulas.aux) )
[  123s] Output written on _formulas.dvi (19 pages, 3244 bytes).
[  123s] Transcript written on _formulas.log.
[  123s] Error: /invalidfileaccess in --run--
[  123s] Operand stack:
[  123s]    (_form0.eps)   (r)
[  123s] Execution stack:
[  123s]    %interp_exit   .runexec2   --nostringval--   run   --nostringval--   2   %stopped_push   --nostringval--   run   run   false   1   %stopped_push   1974   1   3   %oparray_pop   1973   1   3   %oparray_pop   1961   1   3   %oparray_pop   1817   1   3   %oparra
y_pop   --nostringval--   %errorexec_pop   .runexec2   --nostringval--   run   --nostringval--   2   %stopped_push   --nostringval--   1974   1   3   %oparray_pop   run
[  123s] Dictionary stack:
[  123s]    --dict:737/1123(ro)(G)--   --dict:0/20(G)--   --dict:76/200(L)--
[  123s] Current allocation mode is local
[  123s] Last OS error: Permission denied
[  123s] Current file position is 148
[  123s] GPL Ghostscript 9.52: Unrecoverable error, exit code 1
[  123s] error: Problem running ghostscript gs -q -g56x50 -r384x384x -sDEVICE=ppmraw -sOutputFile=_form0.pnm -dNOPAUSE -dBATCH -- _form0.ps. Check your installation!
```

And `make man`or `make docs` goes further than previously, but still fail with no obvious errors, only warnings:
```
[  831s] + make -j1 VERBOSE=1 docs
[  831s] (cd doc; make html)
[  831s] make[1]: Entering directory '/home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc'
[  831s] python3 "source/build_driver_summary.py" "source/drivers/raster" raster_driver_summary "source/drivers/raster/driver_summary.rst"
[  831s] python3 "source/build_driver_summary.py" "source/drivers/vector" vector_driver_summary "source/drivers/vector/driver_summary.rst"
[  831s] python3 "source/build_configoptions_index.py"  "source" "source/user/configoptions_index_generated.rst"
[  831s] sphinx-build -M html "source" "build" --keep-going -j auto -W 
[  832s] Running Sphinx v2.3.1
[  832s] making output directory... done
[  832s] building [mo]: targets for 0 po files that are out of date
[  832s] building [html]: targets for 459 source files that are out of date
[  832s] updating environment: [new config] 459 added, 0 changed, 0 removed
[  832s] reading sources... [  3%] about .. api/gnm_cpp
[  832s] /usr/lib/python3.8/site-packages/sphinx/util/docutils.py:283: RemovedInSphinx30Warning: function based directive support is now deprecated. Use class based directive instead.
[  832s]   warnings.warn('function based directive support is now deprecated. '
[  832s] reading sources... [  6%] api/index .. development/index
[  833s] reading sources... [ 10%] development/rfc/index .. development/rfc/rfc23_ogr_unicode
[  833s] reading sources... [ 13%] development/rfc/rfc24_progressive_data_support .. development/rfc/rfc38_ogr_faster_open
[  833s] reading sources... [ 17%] development/rfc/rfc39_ogr_layer_algebra .. development/rfc/rfc52_strict_sql_quoting
[  833s] reading sources... [ 20%] development/rfc/rfc53_ogr_notnull_default .. development/rfc/rfc66_randomlayerreadwrite
[  833s] reading sources... [ 24%] development/rfc/rfc67_nullfieldvalues .. drivers/raster/Idrisi
[  833s] reading sources... [ 27%] drivers/raster/aaigrid .. drivers/raster/cals
[  833s] reading sources... [ 31%] drivers/raster/ceos .. drivers/raster/doq2
[  833s] reading sources... [ 34%] drivers/raster/driver_summary .. drivers/raster/fit
[  833s] reading sources... [ 37%] drivers/raster/fits .. drivers/raster/gta
[  833s] reading sources... [ 41%] drivers/raster/gtiff .. drivers/raster/isis3
[  833s] reading sources... [ 44%] drivers/raster/jdem .. drivers/raster/lcp
[  833s] reading sources... [ 48%] drivers/raster/leveller .. drivers/raster/ngsgeoid
[  833s] reading sources... [ 51%] drivers/raster/ngw .. drivers/raster/png
[  833s] reading sources... [ 55%] drivers/raster/pnm .. drivers/raster/safe
[  833s] reading sources... [ 58%] drivers/raster/sar_ceos .. drivers/raster/vicar
[  833s] reading sources... [ 62%] drivers/raster/vrt .. drivers/vector/avce00
[  833s] reading sources... [ 65%] drivers/vector/bna .. drivers/vector/eeda
[  833s] reading sources... [ 68%] drivers/vector/elasticsearch .. drivers/vector/gmt
[  833s] reading sources... [ 72%] drivers/vector/gpkg .. drivers/vector/mdb
[  833s] reading sources... [ 75%] drivers/vector/memory .. drivers/vector/ogdi
[  833s] reading sources... [ 79%] drivers/vector/openair .. drivers/vector/segy
[  833s] reading sources... [ 82%] drivers/vector/selafin .. drivers/vector/xls
[  833s] reading sources... [ 86%] drivers/vector/xlsx .. programs/gdal_polygonize
[  833s] reading sources... [ 89%] programs/gdal_proximity .. programs/gdalmove
[  833s] reading sources... [ 93%] programs/gdalsrsinfo .. programs/rgb2pct
[  833s] reading sources... [ 96%] programs/vector_common_options .. user/configoptions
[  833s] reading sources... [100%] user/configoptions_index_generated .. user/virtual_file_systems
[  833s] 
[  833s] waiting for workers...
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/cpl.rst:10: WARNING: Invalid definition: Expected identifier in nested name, got keyword: typedef [error at 13]
[  928s]   const typedef char*(* CPLFileFinder) (const char *, const char *)
[  928s]   -------------^
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:26: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:26: WARNING: Duplicate declaration, virtual std::string OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:47: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:47: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:47: WARNING: Duplicate declaration, OGRErr OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:47: WARNING: Duplicate declaration, virtual std::string OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:68: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:68: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:68: WARNING: Duplicate declaration, OGRErr OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:68: WARNING: Duplicate declaration, virtual std::string OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:75: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:75: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:75: WARNING: Duplicate declaration, OGRErr OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:75: WARNING: Duplicate declaration, virtual std::string OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:82: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:82: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:82: WARNING: Duplicate declaration, OGRErr OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:82: WARNING: Duplicate declaration, virtual std::string OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:89: WARNING: Duplicate declaration, OGRErr OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:89: WARNING: Duplicate declaration, virtual std::string OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:96: WARNING: Duplicate declaration, OGRErr OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:96: WARNING: Duplicate declaration, virtual std::string OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:103: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:103: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:103: WARNING: Duplicate declaration, OGRErr OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:103: WARNING: Duplicate declaration, virtual std::string OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:110: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:110: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:110: WARNING: Duplicate declaration, OGRErr OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:110: WARNING: Duplicate declaration, virtual std::string OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:117: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:117: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:117: WARNING: Duplicate declaration, OGRErr OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:117: WARNING: Duplicate declaration, virtual std::string OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:124: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:124: WARNING: Duplicate declaration, OGRErr OGRGeometry::importFromWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:124: WARNING: Duplicate declaration, OGRErr OGRGeometry::exportToWkt()
[  928s] /home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc/source/api/ogrgeometry_cpp.rst:124: WARNING: Duplicate declaration, virtual std::string OGRGeometry::exportToWkt()
[  928s] looking for now-outdated files... none found
[  929s] pickling environment... done
[  929s] checking consistency... done
[  929s] preparing documents... done
[  929s] writing output... [  3%] about_no_title .. api/ogr_srs_api
[  959s] writing output... [  7%] api/ogrfeature_cpp .. development/rfc/rfc11_fastidentify
[ 1001s] writing output... [ 11%] development/rfc/rfc12_filemanagement .. development/rfc/rfc27_supportdata
[ 1001s] writing output... [ 14%] development/rfc/rfc28_sqlfunc .. development/rfc/rfc42_find_laundered_fields
[ 1001s] writing output... [ 18%] development/rfc/rfc43_getmetadatadomainlist .. development/rfc/rfc58_removing_dataset_nodata_value
[ 1002s] writing output... [ 22%] development/rfc/rfc59.1_utilities_as_a_library .. development/rfc/rfc72_pytest
[ 1003s] writing output... [ 25%] development/rfc/rfc73_proj6_wkt2_srsbarn .. drivers/raster/bag
[ 1003s] writing output... [ 29%] drivers/raster/blx .. drivers/raster/db2
[ 1004s] writing output... [ 33%] drivers/raster/dds .. drivers/raster/envi
[ 1004s] writing output... [ 37%] drivers/raster/epsilon .. drivers/raster/grib
[ 1005s] writing output... [ 40%] drivers/raster/gs7bg .. drivers/raster/iris
[ 1009s] writing output... [ 44%] drivers/raster/isce .. drivers/raster/kro
[ 1010s] writing output... [ 48%] drivers/raster/l1b .. drivers/raster/ndf
[ 1010s] writing output... [ 51%] drivers/raster/netcdf .. drivers/raster/plmosaic
[ 1011s] writing output... [ 55%] drivers/raster/png .. drivers/raster/safe
[ 1011s] writing output... [ 59%] drivers/raster/sar_ceos .. drivers/raster/vrt
[ 1012s] writing output... [ 62%] drivers/raster/vrt_multidimensional .. drivers/vector/cad
[ 1013s] writing output... [ 66%] drivers/vector/carto .. drivers/vector/filegdb
[ 1013s] writing output... [ 70%] drivers/vector/flatgeobuf .. drivers/vector/grass
[ 1014s] writing output... [ 74%] drivers/vector/gtm .. drivers/vector/mssqlspatial
[ 1017s] writing output... [ 77%] drivers/vector/mvt .. drivers/vector/pg
[ 1018s] writing output... [ 81%] drivers/vector/pg_advanced .. drivers/vector/sxf
[ 1019s] writing output... [ 85%] drivers/vector/tiger .. programs/gdal2tiles
[ 1020s] writing output... [ 88%] programs/gdal_calc .. programs/gdalcompare
[ 1020s] writing output... [ 92%] programs/gdaldem .. programs/ogrinfo
[ 1023s] writing output... [ 96%] programs/ogrlineref .. tutorials/vector_api_tut
[ 1024s] writing output... [100%] tutorials/vector_driver_tut .. user/virtual_file_systems
[ 1026s] 
[ 1026s] waiting for workers...
[ 1029s] generating indices...  genindexdone
[ 1029s] writing additional pages...  searchdone
[ 1029s] copying images... [  4%] ../images/OSGeo_project.png
[ 1029s] copying images... [  9%] ../images/foss4g2020.png
[ 1029s] copying images... [ 14%] development/rfc/../../../images/rfc45/rfc_2d_array.png
[ 1029s] copying images... [ 19%] development/rfc/../../../images/rfc45/rfc_tiled.png
[ 1029s] copying images... [ 23%] development/rfc/../../../images/rfc45/rfc_TIP.png
[ 1029s] copying images... [ 28%] development/rfc/../../../images/rfc45/rfc_BIT.png
[ 1029s] copying images... [ 33%] development/rfc/../../../images/rfc45/rfc_BSQ.png
[ 1029s] copying images... [ 38%] development/rfc/../../../images/rfc49/classOGRGeometry.png
[ 1029s] copying images... [ 42%] development/rfc/../../../images/rfc64/classOGRGeometry_RFC64.png
[ 1029s] copying images... [ 47%] drivers/raster/../../../images/jpipsequence.PNG
[ 1029s] copying images... [ 52%] drivers/raster/../../../images/components.PNG
[ 1029s] copying images... [ 57%] drivers/raster/../../../images/gdalsequence.PNG
[ 1029s] copying images... [ 61%] programs/../../images/gdal_viewshed.png
[ 1029s] copying images... [ 66%] tutorials/../../images/grid/gridding.png
[ 1029s] copying images... [ 71%] tutorials/../../images/grid/ellipse.png
[ 1029s] copying images... [ 76%] user/../../images/style_pen1.png
[ 1029s] copying images... [ 80%] user/../../images/style_pen2.png
[ 1029s] copying images... [ 85%] user/../../images/style_pen3.png
[ 1029s] copying images... [ 90%] user/../../images/style_ogr_brush.png
[ 1029s] copying images... [ 95%] user/../../images/style_ogr_sym.png
[ 1029s] copying images... [100%] user/../../images/style_textanchor.png
[ 1029s] 
[ 1029s] copying downloadable files... [100%] contributing/example.txt
[ 1029s] 
[ 1029s] copying static files... ... done
[ 1029s] copying extra files... done
[ 1032s] dumping search index in English (code: en)... done
[ 1033s] dumping object inventory... done
[ 1033s] build finished with problems, 39 warnings.
[ 1033s] /usr/lib/python3.8/site-packages/sphinx_rtd_theme/search.html:21: RemovedInSphinx30Warning: To modify script_files in the theme is deprecated. Please insert a <script> tag directly in your theme instead.
[ 1033s]   {% endblock %}
[ 1036s] make[1]: *** [Makefile:35: html] Error 1
[ 1036s] make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/gdal-3.1.0/doc'
[ 1036s] make: *** [GNUmakefile:167: docs] Error 2
```

--- a/gdal/Doxyfile
+++ b/gdal/Doxyfile
@@ -954,7 +954,12 @@ PREDEFINED             = HAVE_DLFCN_H \
                          HAVE_LONG_LONG=1 \
                          HAVE_SC_PHYS_PAGES=1 \
                          final= \
-                         CPL_RESTRICT=
+                         CPL_RESTRICT= \
+                         CPL_WARN_UNUSED_RESULT= \
+                         CPL_RETURNS_NONNULL= \
+                         CPLSTRING_METHOD_DLL= \
+                         CPL_NO_RETURN= \
+                         EXPERIMENTAL_CPL_WARN_UNUSED_RESULT=
 
 # If the MACRO_EXPANSION and EXPAND_ONLY_PREDEF tags are set to YES then
 # this tag can be used to specify a list of macro names that should be expanded.

--- a/gdal/gcore/gdal.h
+++ b/gdal/gcore/gdal.h
@@ -956,7 +956,11 @@ CPLErr CPL_DLL CPL_STDCALL GDALGetRasterHistogram( GDALRasterBandH hBand,
                                        int nBuckets, int *panHistogram,
                                        int bIncludeOutOfRange, int bApproxOK,
                                        GDALProgressFunc pfnProgress,
-                                       void * pProgressData ) CPL_WARN_DEPRECATED("Use GDALGetRasterHistogramEx() instead");
+                                       void * pProgressData )
+/*! @cond Doxygen_Suppress */
+    CPL_WARN_DEPRECATED("Use GDALGetRasterHistogramEx() instead")
+/*! @endcond */
+    ;
 CPLErr CPL_DLL CPL_STDCALL GDALGetRasterHistogramEx( GDALRasterBandH hBand,
                                        double dfMin, double dfMax,
                                        int nBuckets, GUIntBig *panHistogram,
@@ -968,7 +972,11 @@ CPLErr CPL_DLL CPL_STDCALL GDALGetDefaultHistogram( GDALRasterBandH hBand,
                                        int *pnBuckets, int **ppanHistogram,
                                        int bForce,
                                        GDALProgressFunc pfnProgress,
-                                       void * pProgressData ) CPL_WARN_DEPRECATED("Use GDALGetDefaultHistogramEx() instead");
+                                       void * pProgressData )
+/*! @cond Doxygen_Suppress */
+    CPL_WARN_DEPRECATED("Use GDALGetDefaultHistogramEx() instead")
+/*! @endcond */
+    ;
 CPLErr CPL_DLL CPL_STDCALL GDALGetDefaultHistogramEx( GDALRasterBandH hBand,
                                        double *pdfMin, double *pdfMax,
                                        int *pnBuckets, GUIntBig **ppanHistogram,
@@ -977,7 +985,11 @@ CPLErr CPL_DLL CPL_STDCALL GDALGetDefaultHistogramEx( GDALRasterBandH hBand,
                                        void * pProgressData );
 CPLErr CPL_DLL CPL_STDCALL GDALSetDefaultHistogram( GDALRasterBandH hBand,
                                        double dfMin, double dfMax,
-                                       int nBuckets, int *panHistogram ) CPL_WARN_DEPRECATED("Use GDALSetDefaultHistogramEx() instead");
+                                       int nBuckets, int *panHistogram )
+/*! @cond Doxygen_Suppress */
+    CPL_WARN_DEPRECATED("Use GDALSetDefaultHistogramEx() instead")
+/*! @endcond */
+    ;
 CPLErr CPL_DLL CPL_STDCALL GDALSetDefaultHistogramEx( GDALRasterBandH hBand,
                                        double dfMin, double dfMax,
                                        int nBuckets, GUIntBig *panHistogram );

--- a/gdal/gcore/gdal_priv.h
+++ b/gdal/gcore/gdal_priv.h
@@ -604,7 +604,9 @@ class CPL_DLL GDALDataset : public GDALMajorObject
     CPLErr BuildOverviews( const char *, int, int *,
                            int, int *, GDALProgressFunc, void * );
 
+#ifndef DOXYGEN_XML
     void ReportError(CPLErr eErrClass, CPLErrorNum err_no, const char *fmt, ...)  CPL_PRINT_FUNC_FORMAT (4, 5);
+#endif
 
     char ** GetMetadata(const char * pszDomain = "") override;
 
@@ -612,8 +614,6 @@ class CPL_DLL GDALDataset : public GDALMajorObject
 #ifdef DOXYGEN_SKIP
     CPLErr      SetMetadata( char ** papszMetadata,
                              const char * pszDomain ) override;
-    const char *GetMetadataItem( const char * pszName,
-                                 const char * pszDomain ) override;
     CPLErr      SetMetadataItem( const char * pszName,
                                  const char * pszValue,
                                  const char * pszDomain ) override;
@@ -1252,11 +1252,8 @@ class CPL_DLL GDALRasterBand : public GDALMajorObject
 
 // Only defined when Doxygen enabled
 #ifdef DOXYGEN_SKIP
-    char      **GetMetadata( const char * pszDomain = "" ) override;
     CPLErr      SetMetadata( char ** papszMetadata,
                              const char * pszDomain ) override;
-    const char *GetMetadataItem( const char * pszName,
-                                 const char * pszDomain ) override;
     CPLErr      SetMetadataItem( const char * pszName,
                                  const char * pszValue,
                                  const char * pszDomain ) override;

--- a/gdal/ogr/ogr_geometry.h
+++ b/gdal/ogr/ogr_geometry.h
@@ -421,13 +421,18 @@ class CPL_DLL OGRGeometry
                                 OGRwkbVariant=wkbVariantOldOgc ) const = 0;
     virtual OGRErr importFromWkt( const char ** ppszInput ) = 0;
 
+#ifndef DOXYGEN_XML
     /** Deprecated.
      * @deprecated in GDAL 2.3
      */
-    OGRErr importFromWkt( char ** ppszInput ) CPL_WARN_DEPRECATED("Use importFromWkt(const char**) instead")
+    OGRErr importFromWkt( char ** ppszInput )
+/*! @cond Doxygen_Suppress */
+        CPL_WARN_DEPRECATED("Use importFromWkt(const char**) instead")
+/*! @endcond */
     {
         return importFromWkt( const_cast<const char**>(ppszInput) );
     }
+#endif
 
     OGRErr exportToWkt( char ** ppszDstText,
         OGRwkbVariant=wkbVariantOldOgc ) const;
@@ -881,9 +886,16 @@ class CPL_DLL OGRPoint : public OGRGeometry
     OGRErr exportToWkb( OGRwkbByteOrder, unsigned char *,
                         OGRwkbVariant=wkbVariantOldOgc )
         const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::importFromWkt; /** deprecated */
+#endif
+
     OGRErr importFromWkt( const char ** ) override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a point to WKT
     /// \param opts  Output options.
@@ -1187,9 +1199,16 @@ class CPL_DLL OGRSimpleCurve: public OGRCurve
     virtual OGRErr exportToWkb( OGRwkbByteOrder, unsigned char *,
                                 OGRwkbVariant=wkbVariantOldOgc )
         const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::importFromWkt; /** deprecated */
+#endif
+
     OGRErr importFromWkt( const char ** ) override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a simple curve to WKT
     /// \param opts  Output options.
@@ -1481,9 +1500,16 @@ class CPL_DLL OGRCircularString : public OGRSimpleCurve
     virtual OGRErr exportToWkb( OGRwkbByteOrder, unsigned char *,
                                 OGRwkbVariant=wkbVariantOldOgc )
         const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::importFromWkt; /** deprecated */
+#endif
+
     OGRErr importFromWkt( const char ** ) override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a circular string to WKT
     /// \param opts  Output options.
@@ -1701,9 +1727,16 @@ class CPL_DLL OGRCompoundCurve : public OGRCurve
     virtual OGRErr exportToWkb( OGRwkbByteOrder, unsigned char *,
                                 OGRwkbVariant=wkbVariantOldOgc )
         const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::importFromWkt; /** deprecated */
+#endif
+
     OGRErr importFromWkt( const char ** ) override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a compound curve to WKT
     /// \param opts  Output options.
@@ -1904,9 +1937,16 @@ class CPL_DLL OGRCurvePolygon : public OGRSurface
     virtual OGRErr exportToWkb( OGRwkbByteOrder, unsigned char *,
                                 OGRwkbVariant=wkbVariantOldOgc )
         const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::importFromWkt; /** deprecated */
+#endif
+
     OGRErr importFromWkt( const char ** ) override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a curve polygon to WKT
     /// \param opts  Output options.
@@ -2048,10 +2088,16 @@ class CPL_DLL OGRPolygon : public OGRCurvePolygon
     virtual OGRErr exportToWkb( OGRwkbByteOrder, unsigned char *,
                                 OGRwkbVariant=wkbVariantOldOgc )
         const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::importFromWkt; /** deprecated */
+#endif
+
     OGRErr importFromWkt( const char ** ) override;
 
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a polygon to WKT
     /// \param opts  Output options.
@@ -2238,10 +2284,16 @@ class CPL_DLL OGRGeometryCollection : public OGRGeometry
     virtual OGRErr exportToWkb( OGRwkbByteOrder, unsigned char *,
                                 OGRwkbVariant=wkbVariantOldOgc )
         const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::importFromWkt; /** deprecated */
+#endif
+
     OGRErr importFromWkt( const char ** ) override;
 
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a geometry collection to WKT
     /// \param opts  Output options.
@@ -2341,10 +2393,16 @@ class CPL_DLL OGRMultiSurface : public OGRGeometryCollection
     // Non standard (OGRGeometry).
     virtual const char *getGeometryName() const override;
     virtual OGRwkbGeometryType getGeometryType() const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::importFromWkt; /** deprecated */
+#endif
+
     OGRErr importFromWkt( const char ** ) override;
 
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a geometry collection to WKT
     /// \param opts  Output options.
@@ -2439,7 +2497,10 @@ class CPL_DLL OGRMultiPolygon : public OGRMultiSurface
     // Non-standard (OGRGeometry).
     virtual const char *getGeometryName() const override;
     virtual OGRwkbGeometryType getGeometryType() const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a multipolygon to WKT
     /// \param opts  Output options.
@@ -2538,9 +2599,16 @@ class CPL_DLL OGRPolyhedralSurface : public OGRSurface
     virtual OGRErr exportToWkb( OGRwkbByteOrder, unsigned char *,
                                 OGRwkbVariant=wkbVariantOldOgc )
         const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::importFromWkt; /** deprecated */
+#endif
+
     OGRErr importFromWkt( const char ** ) override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a polyhedral surface to WKT
     /// \param opts  Output options.
@@ -2718,9 +2786,16 @@ class CPL_DLL OGRMultiPoint : public OGRGeometryCollection
     // Non-standard (OGRGeometry).
     virtual const char *getGeometryName() const override;
     virtual OGRwkbGeometryType getGeometryType() const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::importFromWkt; /** deprecated */
+#endif
+
     OGRErr importFromWkt( const char ** ) override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a multipoint to WKT
     /// \param opts  Output options.
@@ -2803,9 +2878,16 @@ class CPL_DLL OGRMultiCurve : public OGRGeometryCollection
     // Non standard (OGRGeometry).
     virtual const char *getGeometryName() const override;
     virtual OGRwkbGeometryType getGeometryType() const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::importFromWkt; /** deprecated */
+#endif
+
     OGRErr importFromWkt( const char ** ) override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a multicurve to WKT
     /// \param opts  Output options.
@@ -2884,7 +2966,10 @@ class CPL_DLL OGRMultiLineString : public OGRMultiCurve
     // Non standard (OGRGeometry).
     virtual const char *getGeometryName() const override;
     virtual OGRwkbGeometryType getGeometryType() const override;
+
+#ifndef DOXYGEN_XML
     using OGRGeometry::exportToWkt;
+#endif
 
     /// Export a multilinestring to WKT
     /// \param opts  Output options.

--- a/gdal/ogr/ogr_spatialref.h
+++ b/gdal/ogr/ogr_spatialref.h
@@ -293,7 +293,11 @@ class CPL_DLL OGRSpatialReference
                                       const char *pszUnitAuthority = nullptr,
                                       const char *pszUnitCode = nullptr);
 
-    double      GetLinearUnits( char ** ) const CPL_WARN_DEPRECATED("Use GetLinearUnits(const char**) instead");
+    double      GetLinearUnits( char ** ) const
+/*! @cond Doxygen_Suppress */
+        CPL_WARN_DEPRECATED("Use GetLinearUnits(const char**) instead")
+/*! @endcond */
+        ;
     double      GetLinearUnits( const char ** = nullptr ) const;
 /*! @cond Doxygen_Suppress */
     double      GetLinearUnits( std::nullptr_t ) const
@@ -302,7 +306,10 @@ class CPL_DLL OGRSpatialReference
 
     double      GetTargetLinearUnits( const char *pszTargetKey,
                                       char ** ppszRetName ) const
-            CPL_WARN_DEPRECATED("Use GetTargetLinearUnits(const char*, const char**)");
+/*! @cond Doxygen_Suppress */
+            CPL_WARN_DEPRECATED("Use GetTargetLinearUnits(const char*, const char**)")
+/*! @endcond */
+            ;
     double      GetTargetLinearUnits( const char *pszTargetKey,
                                       const char ** ppszRetName = nullptr ) const;
 /*! @cond Doxygen_Suppress */
@@ -311,14 +318,22 @@ class CPL_DLL OGRSpatialReference
 /*! @endcond */
 
     OGRErr      SetAngularUnits( const char *pszName, double dfInRadians );
-    double      GetAngularUnits( char ** ) const CPL_WARN_DEPRECATED("Use GetAngularUnits(const char**) instead");
+    double      GetAngularUnits( char ** ) const
+/*! @cond Doxygen_Suppress */
+        CPL_WARN_DEPRECATED("Use GetAngularUnits(const char**) instead")
+/*! @endcond */
+        ;
     double      GetAngularUnits( const char ** = nullptr ) const;
 /*! @cond Doxygen_Suppress */
     double      GetAngularUnits( std::nullptr_t ) const
         { return GetAngularUnits( static_cast<const char**>(nullptr) ); }
 /*! @endcond */
 
-    double      GetPrimeMeridian( char ** ) const CPL_WARN_DEPRECATED("Use GetPrimeMeridian(const char**) instead");
+    double      GetPrimeMeridian( char ** ) const
+/*! @cond Doxygen_Suppress */
+        CPL_WARN_DEPRECATED("Use GetPrimeMeridian(const char**) instead")
+/*! @endcond */
+        ;
     double      GetPrimeMeridian( const char ** = nullptr ) const;
 /*! @cond Doxygen_Suppress */
     double      GetPrimeMeridian( std::nullptr_t ) const

--- a/gdal/port/cpl_conv.h
+++ b/gdal/port/cpl_conv.h
@@ -178,7 +178,7 @@ const char CPL_DLL *CPLLaunderForFilename(const char* pszName,
 /* -------------------------------------------------------------------- */
 
 /** Callback for CPLPushFileFinder */
-typedef const char *(*CPLFileFinder)(const char *, const char *);
+typedef char const *(*CPLFileFinder)(const char *, const char *);
 
 const char    CPL_DLL *CPLFindFile(const char *pszClass,
                                    const char *pszBasename);

